src/components/fx/hover.js

```diff
@@ -689,7 +689,7 @@ function createHoverText(hoverData, opts, gd) {
             // strip out our pseudo-html elements from d.name (if it exists at all)
             name = svgTextUtils.plainText(d.name || '');
 
-            if(name.length > 15) name = name.substr(0, 12) + '...';
+            if(name.length > commonLabelOpts.namelength && commonLabelOpts.namelength > 0) name = name.substr(0, commonLabelOpts.namelength - 3) + '...';
```

Comment:
Too bad the JS token `Infinity` isn't JSON serializable. I'm ok with making `namelength: 0` means no truncation, even though it feels contradictory to me.

Perhaps we should instead add support for an `Infinity`-like value in `valType: 'number'` attributes. For example, we could make `namelength: null` correspond to `namelength === Infinity`. Though, I can't think of any other attributes that could benefit from it, so I'm not sure it's worth the time.

cc @alexcjohnson